src/components/planner/OptionsController.tsx

```diff
@@ -1,148 +1,186 @@
+import { ThemeContext } from '../../contexts/ThemeContext'
+import {
+  PencilAltIcon,
+} from '@heroicons/react/solid'
+
+import { useContext } from 'react';
 
 type Props = {
   multipleOptionsHook: [MultipleOptions, React.Dispatch<React.SetStateAction<MultipleOptions>>]
 }
 
+interface Option {
+  id: number;
+  icon: string;
+  name: string;
+}
+
+const Option = ({item, selectedHook, setOptionIndex, multipleOptionsHook}) => {
+  const [selected, setSelected] = selectedHook   
+  return (
+      <div 
+          onClick={() => {
+            setSelected(item.id);
+            setOptionIndex(item.id);
+          }} 
+          className={`box-border p-2 w-10 h-10 aspect-square rounded cursor-pointer border-2 border-transparent hover:border-primary/75 hover:dark:border-primary/50 dark:shadow transition-colors ease-in-out delay-150 ${selected === item.id ? "text-white bg-primary/75 dark:bg-primary/50" : "bg-lightish dark:bg-darkish"}`}
+      >
+          {item.icon}
+      </div>
+  )
+}
+
 const OptionsController = ({ multipleOptionsHook }: Props) => {
   const [multipleOptions, setMultipleOptions] = multipleOptionsHook
   const optionIndexes = Array.from({ length: 10 }, (_, i) => i)
-  const [isEditingName, setIsEditingName] = useState(false)
-  const menuButtonRef = useRef<HTMLButtonElement>(null)
 
-  const setNextOptionIndex = () => {
+  const {enabled, setEnabled} = useContext(ThemeContext)
 
-    setMultipleOptions((prev) => ({
-      index: prev.index + 1,
-      selected: prev.options[prev.index + 1],
-      options: [...prev.options],
-      names: prev.names
-    }))
-  }
+  const [selected, setSelected] = useState(() => {
+    const selectedOption = localStorage.getItem("niaefeup-tts.selected-option");
 
+    if(!selectedOption) {
+      return 1;
+    } else {
+      return parseInt(selectedOption);
+    }
+  });
 
-  const setPreviousOptionIndex = () => {
-    setMultipleOptions((prev) => ({
-      index: prev.index - 1,
-      selected: prev.options[prev.index - 1],
-      options: [...prev.options],
-      names: prev.names
-    }))
-  }
+  // Update local storage value when user selects a new option
+  useEffect(() => {
+    localStorage.setItem("niaefeup-tts.selected-option", selected.toString())
+  }, [selected])
+
+  const [optionsList, setOptionsList] = useState(() => {
+    const optionsList: Array<Option> = JSON.parse(localStorage.getItem("niaefeup-tts.optionsList"));
+
+    if(!optionsList) {
+      return [
+        { id: 1, icon: "😊", name: "Horário 1" },
+        { id: 2, icon: "🌟", name: "Horário 2" },
+        { id: 3, icon: "🚀", name: "Horário 3" },
+        { id: 4, icon: "📚", name: "Horário 4" },
+        { id: 5, icon: "🎉", name: "Horário 5" },
+        { id: 6, icon: "💻", name: "Horário 6" },
+        { id: 7, icon: "🌈", name: "Horário 7" },
+        { id: 8, icon: "🍀", name: "Horário 8" },
+        { id: 9, icon: "🎓", name: "Horário 9" },
+        { id: 10, icon: "🤖", name: "Horário 10" }
```

Comment:
We want the emojis hard coded?
If not, this code can be reduced to one line of a list comprehension